docs/book/src/getting-started.md

```diff
@@ -206,9 +206,9 @@ return ctrl.Result{RequeueAfter: nextRun.Sub(r.Now())}, nil
 
 #### In the context of our example
 
-When a Custom Resource is applied to the cluster, there's a designated controller to manage the Memcached Kind. You can check its reconciliation implemented:
+When a Custom Resource is applied to the cluster, there's a designated controller to manage the Memcached Kind. You can how its reconciliation is implemented:
 
-From `testdata/project-v4-with-deploy-image/internal/controller/memcached_controller.go`:
+From `internal/controller/memcached_controller.go`:
```

Comment:
^ it is the testdata dir of the project where we add the examples

Reply:
Yes, but as mentioned above, that content is different than these instructions result in. The locally generated content from following the commands above are under the updated path. If we want to refer to the git repo for additional examples I think that could work. But as this part is, it can be confusing referring to a path that is not part of the local content.

Reply:
I see; we should actually generate the sample under the docs ( with the same instructions ) as we do for the tutorials and just use the code directly from the sample.

So, that when we run `make generate` we will be able to keep this one update. 

See: 
- Here we generate the sample: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/cronjob-tutorial/testdata/project
- Then, you can check that in the doc we are ref, the code directly like: `{{#literatego ./testdata/emptycontroller.go}}` in https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/controller-overview.md
- You can either check the make generate target and see that we build the samples using https://github.com/kubernetes-sigs/kubebuilder/tree/master/hack/docs

Otherwise, it is hard to keep it maintained.

Reply:
+1 to as @camilamacedo86 said. If we could directly refer the paths in the doc, that would make it easier for us to maintain it. Huge + for the idea though. We would have to probably discuss more on the how the testate would be structured to refer multi-version/multi-group scenarios for the same plugins as we have today.

Reply:
I looked in to using `literatego` and making this more integrated, but there were some issues with that approach. The biggest challenge is that pulls in the entire source file, when in most cases here we only want to highlight a small portion of it.

I hope I've reached a happy medium. I do now generate `testdata` for the getting started docs, and I try to directly pull in relevant content where possible. For the other instances, I just updated the snippet of code the docs were highlighting, but also added a link to the file in the repo so the user has a canonical reference for anything being covered. Then if they do diverge at some point, it's at least possible to follow the link to see the current full example that should match what they get locally by running the commands.

Let me know if there are any better ideas or other ways to improve that.